src/loader/bgw_launcher.c

```diff
@@ -481,21 +504,22 @@ message_restart_action(HTAB *db_htab, BgwMessage *message, VirtualTransactionId
 		terminate_background_worker(db_he->db_scheduler_handle);
 		wait_for_background_worker_shutdown(db_he->db_scheduler_handle);
 	}
-	else if (!bgw_total_workers_increment())	/* we still need to increment
-												 * if we haven't found an
-												 * entry */
+	else
 	{
-		report_bgw_limit_exceeded();
-		bgw_message_send_ack(message, ACK_FAILURE);
-		return;
+		if (!bgw_total_workers_increment()) /* we still need to increment if
+											 * we haven't found an entry */
+		{
+			report_bgw_limit_exceeded();
+			bgw_message_send_ack(message, ACK_FAILURE);
+			return;
+		}
+		else
+			db_he->incremented = true;
```

Comment:
Questionable Nit: since `incremented` must be true, we can set it unconditionally before `register_entrypoint_for_db` and use the same `if else` from before.

Reply:
I took out the `else` (as above), but still don't think it should be set unconditionally because maybe that might mask some bug.